Feat tre enhancements

narendran1560 opened the pull request:
Issue #, if available:

Description of changes:
Add the SWB-TRE enhancement feature 2 and feature 3

Feature 2
Administrators should not have access to all the researcher workspace
Administrators should have access to the workspace owned by them
This feature would be configurable using
“restrictAdminWorkspaceConnection" flag

Study permission API should not accept application admin as a study admin.
This feature would be configurable using the “disableAdminBYOBSelfAssignment” flag.

Feature 3
SWB researchers were not able to create a study/org.
Disabling the create study button in the UI screen also restricts the backend api for the create study.
This feature would be configurable using
“disableStudyUploadByResearcher" flag.

SWB researchers were not able to upload files in the created study/org.
This feature would be configurable using
“disableStudyUploadByResearcher" flag.

Checklist:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully tested with your changes locally?
- [ ] If new dependencies have been added, have they been pinned to specific versions?
- [ ] Is this change also required on the AWS Solution version?
- [ ] Have you updated openapi.yaml if you made updates to API definition (including add, delete or update parameter and request data schema)?
- [ ] If you had to run manual tests, have you considered automating those tests by adding them to [end-to-end tests](../main/end-to-end-tests/README.md)?
- [ ] If you are updating the changelog and vending out a new release, have you updated versionNumber and versionDate in [.defaults.yml](../main/config/settings/.defaults.yml)

<!-- For major releases please provide internal ticket id -->

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

narendran1560 committed:
SWB-TRE Feature 2 and Feature 3

narendran1560 committed:
SWB-TRE Feature 2 and Feature 3

narendran1560 committed:
Resolved the eslint issue

narendran1560 committed:
Resolved the eslint issue

narendran1560 committed:
Resolved the JS-Lint issues

narendran1560 committed:
Resolved the JS-Lint issues

SanketD92 committed:
chore: vulnerable dependency version fix (#976)

* chore: fix ansi-regex version in lock

narendran1560 committed:
Feature 2 and 3 code review changes added

narendran1560 committed:
Feature 2 and 3 code review changes added

narendran1560 committed:
Merge remote-tracking branch 'upstream/feat-tre-enhancements' into feat-feature-2and3

narendran1560 committed:
resolved the es-lint issues

narendran1560 committed:
resolved the es-lint issues

narendran1560 committed:
resolved the eslint issue

narendran1560 committed:
resolved the eslint issue

narendran1560 committed:
Merge pull request #64 from narendran1560/feat-feature-2and3

SWB-TRE Feature 2 and Feature 3

narendran1560 committed:
Added the feature flag explaination in the example.yml file

narendran1560 committed:
Merge pull request #70 from narendran1560/feat-feature-2and3

Added the feature flag explaination in the example.yml file

narendran1560 committed:
Merge pull request #71 from RLOpenCatalyst/feat-feature-2and3

SWB-TRE Feature 2 and Feature 3

SanketD92 commented on addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/data-sources/register/InputStep.js:
```diff
@@ -72,6 +72,7 @@ class InputStep extends React.Component {
     const result = [];
     _.forEach(list, user => {
       if (!user.isActive) return;
+      if ((user.isAdmin || user.userRole === 'admin') && disableAdminBYOBSelfAssignment === true) return;
```
Aren't these the same check:
`(user.isAdmin || user.userRole === 'admin')`

narendran1560 replied:
Yes. We also feel the same but it was existing code. So we leave it as it is.

SanketD92 commented on addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentButtons.js:
```diff
@@ -137,6 +141,23 @@ class ScEnvironmentButtons extends React.Component {
+    //  Restrict the admin to access workspaces only created by them.
+    if (restrictAdminWorkspaceConnection === true) {
+      if (isAdmin === true) {
+        restrictWorkspaceConnection = false;
```
Let's use the `canConnect` boolean instead of creating a new boolean value.

SanketD92 replied:
`allowWorkspaceConnection` would work better than `restrictWorkspaceConnection`.

SanketD92 commented on addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-connection-service.test.js:
```diff
@@ -635,4 +635,49 @@ jM0re//6SUWx/9VfBLN+6Ul8wcqGR2uCmK/PJpzWYxz0IzhnyA==
     });
   });
+
+  describe('verifyAccess', () => {
```
Could you also add a test case where admin owns the workspace and the `restrictAdminWorkspaceConnection` flag is set to true?

narendran1560 replied:
Added as you suggested.

narendran1560 committed:
Resolved aws-swb PR comments

dependabot[bot] committed:
chore(deps): bump cross-fetch from 3.1.4 to 3.1.5 in /docs (#982)

* chore(deps): bump cross-fetch from 3.1.4 to 3.1.5 in /docs

Bumps [cross-fetch](https://github.com/lquixada/cross-fetch) from 3.1.4 to 3.1.5.
- [Release notes](https://github.com/lquixada/cross-fetch/releases)
- [Commits](https://github.com/lquixada/cross-fetch/compare/v3.1.4...v3.1.5)

---
updated-dependencies:
- dependency-name: cross-fetch
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore: trigger build

* fix: tag pnpm version for github pipeline

* fix: use same pnpm tagged version in pipelines

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Sanket Dharwadkar <sdharwad@amazon.com>

SanketD92 committed:
fix: create log group for flow logs (#984)

rsmayda commented on addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-connection-service.js:
```diff
@@ -124,6 +129,20 @@ class EnvironmentScConnectionService extends Service {
+      this.settings.getBoolean(settingKeys.restrictAdminWorkspaceConnection) || false;
+    const uid = _.get(requestContext, 'principalIdentifier.uid');
+    const isAdmin = _.get(requestContext, 'principal.isAdmin');
+    if (restrictAdminWorkspaceConnection !== true) {
```
```suggestion
    if (!restrictAdminWorkspaceConnection) {
```

narendran1560 replied:
Modified the changes. Resolved

rsmayda commented on addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-connection-service.js:
```diff
@@ -124,6 +129,20 @@ class EnvironmentScConnectionService extends Service {
+    const uid = _.get(requestContext, 'principalIdentifier.uid');
+    const isAdmin = _.get(requestContext, 'principal.isAdmin');
+    if (restrictAdminWorkspaceConnection !== true) {
+      return true;
```
Should we also check to see if user is an admin prior to returning true?

narendran1560 replied:
Yes. As you suggested added in the prior returning true.

rsmayda commented on addons/addon-base-raas/packages/base-raas-services/lib/study/study-service.js:
```diff
@@ -284,7 +285,15 @@ class StudyService extends Service {
+    const disableStudyUploadByResearcher =
+      this.settings.getBoolean(settingKeys.disableStudyUploadByResearcher) || false;
+
+    if (disableStudyUploadByResearcher === true && !isAdmin(requestContext)) {
```
```suggestion
    if (disableStudyUploadByResearcher && !isAdmin(requestContext)) {
```

narendran1560 replied:
Modified the changes. Resolved

rsmayda commented on addons/addon-base-raas/packages/base-raas-services/lib/study/study-service.js:
```diff
@@ -284,7 +285,15 @@ class StudyService extends Service {
+    if (disableStudyUploadByResearcher === true && !isAdmin(requestContext)) {
+      throw this.boom.forbidden('Only admin are authorized to create studies.', true);
+    }
+    if (disableStudyUploadByResearcher !== true && !(isInternalResearcher(requestContext) || isAdmin(requestContext))) {
```
```suggestion
    if (!disableStudyUploadByResearcher && !(isInternalResearcher(requestContext) || isAdmin(requestContext))) {
```

narendran1560 replied:
Modified the changes. Resolved

rsmayda commented on addons/addon-base-raas/packages/base-raas-services/lib/study/study-service.js:
```diff
@@ -632,6 +641,17 @@ class StudyService extends Service {
+    const disableStudyUploadByResearcher =
+      this.settings.getBoolean(settingKeys.disableStudyUploadByResearcher) || false;
+
+    if (disableStudyUploadByResearcher === true && !isAdmin(requestContext)) {
```
```suggestion
    if (disableStudyUploadByResearcher && !isAdmin(requestContext)) {
```

narendran1560 replied:
Modified the changes. Resolved

rsmayda commented on addons/addon-base-raas/packages/base-raas-services/lib/study/study-service.js:
```diff
@@ -632,6 +641,17 @@ class StudyService extends Service {
+    if (disableStudyUploadByResearcher === true && !isAdmin(requestContext)) {
+      throw this.boom.forbidden('Only admin are authorized to upload files.', true);
+    }
+    if (disableStudyUploadByResearcher !== true && !(isInternalResearcher(requestContext) || isAdmin(requestContext))) {
```
```suggestion
    if (!disableStudyUploadByResearcher && !(isInternalResearcher(requestContext) || isAdmin(requestContext))) {
```

narendran1560 replied:
Modified the changes. Resolved

narendran1560 committed:
Merge remote-tracking branch 'upstream/develop' into feat-tre-enhancements

narendran1560 committed:
Merge branch 'feat-tre-enhancements' of https://github.com/RLOpenCatalyst/service-workbench-on-aws into feat-tre-enhancements

narendran1560 committed:
Resolved PR comments which was rasied by Robert

narendran1560 committed:
Merge remote-tracking branch 'upstream/feat-tre-enhancements' into feat-feature-2and3

narendran1560 committed:
Merge pull request #73 from narendran1560/feat-feature-2and3

Resolved aws-swb PR comments

narendran1560 committed:
Merge branch 'feat-tre-enhancements' into feat-tre-enhancements
